src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -762,32 +759,32 @@ InferenceProfiler::Measure(
 
   TimestampVector current_timestamps;
   RETURN_IF_ERROR(manager_->SwapTimestamps(current_timestamps));
+  all_timestamps_.insert(
```

Comment:
This is another notable change. Basically, there is this new "primary" vector holding all timestamps (requests). The code that checks whether requests are valid always starts from whatever's at the beginning of this new primary vector. Rather than creating a vector for each pass. This allows us to not drop requests that might finish between passes.